cell_classification/train.py

```diff
@@ -0,0 +1,149 @@
+from cProfile import label
+import os
+import argparse
+import tensorflow as tf
+import toml
+from augmentation_pipeline import prepare_tf_aug, py_aug, get_augmentation_pipeline
+from segmentation_data_prep import parse_dict, feature_description
+from deepcell.model_zoo.panopticnet import PanopticNet
+from tensorflow.keras.optimizers import SGD, Adam
+from deepcell.utils.train_utils import rate_scheduler, get_callbacks, count_gpus
+from deepcell import losses
+from semantic_head import create_semantic_head
+
+
+class Trainer:
+    """Builds, trains and writes validation metrics for models"""
+
+    def __init__(self, params):
+        """Initialize the trainer with the parameters from the config file
+        Args:
+            params (dict): Dictionary of parameters from the config file
+        """
+        self.params = params
+
+    def prep_data(self):
+        """Prepares training and validation data"""
+        # make datasets and splits
+        dataset = tf.data.TFRecordDataset(self.params["record_path"])
+        dataset = dataset.map(lambda x: tf.io.parse_single_example(x, feature_description))
+        dataset = dataset.map(parse_dict)
+
+        # split into train and validation
+        self.validation_dataset = dataset.take(self.params["num_validation"])
+        self.train_dataset = dataset.skip(self.params["num_validation"])
+
+        # shuffle, batch and augment the training data
+        self.train_dataset = self.train_dataset.shuffle(self.params["shuffle_buffer_size"]).batch(
+            self.params["batch_size"]
+        )
+        augmentation_pipeline = get_augmentation_pipeline(self.params)
+        tf_aug = prepare_tf_aug(augmentation_pipeline)
+        self.train_dataset = self.train_dataset.map(lambda x: py_aug(x, tf_aug))
+        self.train_dataset = self.train_dataset.map(self.prep_batches)
+        self.validation_dataset = self.validation_dataset.batch(self.params["batch_size"])
+        self.validation_dataset = self.validation_dataset.map(self.prep_batches)
+
+    def prep_model(self):
+        """Prepares the model for training"""
+        self.optimizer = Adam(learning_rate=self.params["lr"], clipnorm=0.001)
+        self.lr_sched = rate_scheduler(lr=self.params["lr"], decay=0.99)
+
+        if "test" in self.params.keys() and self.params["test"]:
```

Comment:
Since we may try out different models, probably makes more sense to have unique names for each, that way we can just add on to them.

Reply:
Good idea. I just added this single layer test model here so that we can test if the training loop works. I think we'll probably have a function that takes the params dict and returns a model at a later stage.

Reply:
I'd add such a function at a later stage.